tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -747,13 +719,15 @@ class CudnnConvolutionDescriptor {
 #endif
   }
 
-  void set_use_tensor_op_math(bool use_tensor_op_math) const {
+  void set_use_tensor_op_math(bool use_tensor_op_math) {
 #if CUDNN_VERSION >= 7000
     cudnnMathType_t math_type =
+#if CUDNN_VERSION >= 8000
+        (use_tensor_op_math ? CUDNN_DEFAULT_MATH : CUDNN_FMA_MATH);
```

Comment:
I don't see any references to `CUDNN_FMA_MATH` in the cuDNN 8 documentation. Maybe I don't have the latest documentation? Does `CUDNN_DEFAULT_MATH` now use tensor cores by default? If so, why doesn't `CUBLAS_DEFAULT_MATH` do the same?

Reply:
In CUDNN 8 and CUDA 11, CUDNN_DEFAULT_MATH and CUBLAS_DEFAULT_MATH now both use TensorCores for fp16 operations.

For fp32 they differ because CUBLAS is used extensively in HPC applications that require full fp32 precision. So for FP32, cublas is opt-in for TensorCores while cudnn is opt-out.

Reply:
Good catch. This is wrong. The behavior of CUDNN_DEFAULT_MATH is as you say for float16 inputs, but DOES enable tensor cores for float32.